src/ProjNavigator/sources_form.cpp

```diff
@@ -290,23 +294,55 @@ void SourcesForm::SlotRemoveFileSet() {
 }
 
 void SourcesForm::SlotRemoveFile() {
-  QTreeWidgetItem *item = m_treeSrcHierachy->currentItem();
-  if (item == nullptr) return;
+  // Get all selections
+  auto selectedItems = m_treeSrcHierachy->selectedItems();
+  // Use the r-clicked item for determining what item type we are working with
+  QTreeWidgetItem *refItem = m_treeSrcHierachy->currentItem();
+  auto itemType = refItem->data(0, Qt::WhatsThisPropertyRole);
+
+  // Bail on no selection
+  if (refItem == nullptr) return;
+  if (!selectedItems.count()) return;
+
+  // Track string names for confirmation dialog
+  QStringList files;
```

Comment:
this is duplication of 'selections'. Same data with same condition goes to both containers. Right?

Reply:
Yeah this is just to simplify the creation of the confirmation message since i'd need a way to split out a stringlist (so i can use join) later anyways.